tls.c

```diff
@@ -61,9 +61,38 @@ xwc_end_chain(const br_x509_class **ctx)
     if (err)
         return err;
 
-    for (int i = 0; i < mini_cc->num_name_elts; i++)
+    int i, k;
+    bool allowed = true;
+
+    if (nasp_cc->socket_context->allowed_spiffe_ids_length > 0)
+    {
+        allowed = false;
+    }
+
+    pr_info("nasp: allowed spiffe id len[%d]", nasp_cc->socket_context->allowed_spiffe_ids_length);
+
+    for (i = 0; i < mini_cc->num_name_elts; i++)
     {
-        printk("nasp: peer certificate name_elts[%d]: status: %d, value: %s", i, mini_cc->name_elts[i].status, mini_cc->name_elts[i].buf);
+        printk("nasp: peer certificate name_elts[%d]: status: %d, value: %s (len: %d)", i, mini_cc->name_elts[i].status, mini_cc->name_elts[i].buf, mini_cc->name_elts[i].len);
+        if (mini_cc->name_elts[i].oid == OID_uniformResourceIdentifier)
+        {
+            for (k = 0; k < nasp_cc->socket_context->allowed_spiffe_ids_length; k++)
+            {
+                if (strncmp(nasp_cc->socket_context->allowed_spiffe_ids[k], mini_cc->name_elts[i].buf, strlen(nasp_cc->socket_context->allowed_spiffe_ids[k])) == 0)
+                {
+                    pr_info("nasp: %s == ^%s", mini_cc->name_elts[i].buf, nasp_cc->socket_context->allowed_spiffe_ids[k]);
+                    allowed = true;
+                    break;
+                }
+            }
+        }
+    }
+
+    if (!allowed)
+    {
+        pr_info("nasp: peer certificate is not allowed!");
+
+        return 62;
```

Comment:
What is 62?

Reply:
This is an error code:
/** \brief X.509 status: chain could not be linked to a trust anchor. */
#define BR_ERR_X509_NOT_TRUSTED           62